tokio/tests/join_handle_panic.rs

```diff
@@ -1,5 +1,5 @@
 #![warn(rust_2018_idioms)]
-#![cfg(feature = "full")]
+#![cfg(all(feature = "full", not(target_os = "wasi")))]
 
 struct PanicsOnDrop;
 
```

Comment:
This test should run on WASM too. (Unless catching panics is somehow unsupported on WASM?)